src/net/tcp/listener.rs

```diff
@@ -248,3 +248,21 @@ impl FromRawFd for TcpListener {
         TcpListener::from_std(FromRawFd::from_raw_fd(fd))
     }
 }
+
+impl From<TcpListener> for net::TcpListener {
+    fn from(listener: TcpListener) -> Self {
+        // Safety: This is safe since we are extracting the raw fd from a well-constructed
+        // mio::net::TcpListener which ensures that we actually pass in a valid file
+        // descriptor/socket
+        unsafe {
+            #[cfg(any(unix, target_os = "hermit"))]
```

Comment:
This currently fails to build on WASI. Can you try the following, to see if that fixes it? You might need to add `use std::os::wasi::io::{FromRawFd, IntoRawFd};` to some files.

```suggestion
            #[cfg(any(unix, target_os = "hermit", target_os = "wasi"))]
```

Reply:
Sure. With that addition the lib can be built for the target `wasm32-wasi` on my machine.